lib/puppet/type/cisco_interface_hsrp_group.rb

```diff
@@ -0,0 +1,370 @@
+    desc "Specifies time to wait for IP redundancy clients.
+          Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property preempt_delay_sync
+
+  newproperty(:priority) do
+    desc "Sets Priority value for this hsrp group.
+          Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property priority
+
+  newproperty(:priority_forward_thresh_lower) do
+    desc "Sets Priority forwarding lower threshold value.
+          Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property priority_forward_thresh_lower
+
+  newproperty(:priority_forward_thresh_upper) do
+    desc "Sets Priority forwarding upper threshold value.
+          Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property priority_forward_thresh_upper
+
+  newproperty(:timers_hello_msec) do
+    desc 'Specify hello interval in milliseconds'
+
+    newvalues(:true, :false, :default)
+  end # property timers_hello_msec
+
+  newproperty(:timers_hold_msec) do
+    desc 'Specify hold interval in milliseconds'
+
+    newvalues(:true, :false, :default)
+  end # property timers_hold_msec
+
+  newproperty(:timers_hello) do
+    desc "Sets hello interval. Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property timers_hello
+
+  newproperty(:timers_hold) do
+    desc "Sets hold interval. Valid values are integer, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property timers_hold
+
+  def my_atype
+    self[:authentication_auth_type].nil? ||
+      self[:authentication_auth_type] == :default ||
+      self[:authentication_auth_type] == :cleartext
+  end
+
+  def my_enc
+    self[:authentication_enc_type].nil? ||
+      self[:authentication_enc_type] == :default ||
+      self[:authentication_enc_type] == '0'
+  end
+
+  def my_key
+    self[:authentication_key_type].nil? ||
+      self[:authentication_key_type] == :default ||
+      self[:authentication_key_type] == :'key-chain'
+  end
+
+  def my_compat
+    self[:authentication_compatibility].nil? ||
+      self[:authentication_compatibility] == :default ||
+      self[:authentication_compatibility] == :false
+  end
+
+  def my_str
+    self[:authentication_string].nil? ||
+      self[:authentication_string] == :default ||
+      self[:authentication_string].empty?
+  end
+
+  def my_timeout
+    self[:authentication_timeout].nil? ||
+      self[:authentication_timeout] == :default ||
+      self[:authentication_timeout].zero?
+  end
+
+  def check_auth_str
+    return unless my_str
+    fail ArgumentError, 'All authentication parameters MUST be default when authentication_string is default' unless
```

Comment:
I think it would be more user friendly to list the properties that must be set explicitly instead of stating `All authentication parameters`.  I understand that this is more verbose, but will lessen the confusion.